aws/resource_aws_flow_log.go

```diff
@@ -65,34 +92,13 @@ func resourceAwsFlowLog() *schema.Resource {
 func resourceAwsLogFlowCreate(d *schema.ResourceData, meta interface{}) error {
 	conn := meta.(*AWSClient).ec2conn
 
-	types := []struct {
-		ID   string
-		Type string
-	}{
-		{ID: d.Get("vpc_id").(string), Type: "VPC"},
-		{ID: d.Get("subnet_id").(string), Type: "Subnet"},
-		{ID: d.Get("eni_id").(string), Type: "NetworkInterface"},
-	}
-
-	var resourceId string
-	var resourceType string
-	for _, t := range types {
-		if t.ID != "" {
-			resourceId = t.ID
-			resourceType = t.Type
-			break
-		}
-	}
-
-	if resourceId == "" || resourceType == "" {
-		return fmt.Errorf("Error: Flow Logs require either a VPC, Subnet, or ENI ID")
-	}
-
 	opts := &ec2.CreateFlowLogsInput{
 		DeliverLogsPermissionArn: aws.String(d.Get("iam_role_arn").(string)),
+		LogDestination:           aws.String(d.Get("log_destination").(string)),
+		LogDestinationType:       aws.String(d.Get("log_destination_type").(string)),
 		LogGroupName:             aws.String(d.Get("log_group_name").(string)),
-		ResourceIds:              []*string{aws.String(resourceId)},
-		ResourceType:             aws.String(resourceType),
+		ResourceIds:              aws.StringSlice(d.Get("resource_id").([]string)),
```

Comment:
As `resource_id` is declared as `schema.TypeSet`, `d.Get("resource_id")` is I think of type `*schema.Set`, so some manipulation is required to pass to `aws.StringSlice`.